lxd/api_cluster.go

```diff
@@ -523,6 +523,50 @@ func clusterPutJoin(d *Daemon, req api.ClusterPut) Response {
 			return err
 		}
 
+		// Re-use the client handler and import the images from the leader node which
+		// owns all available images to the joined node
+		go func() {
+			leader, err := d.gateway.LeaderAddress()
+			if err != nil {
+				logger.Errorf("Failed to get current leader node: %v", err)
+				return
+			}
+			var nodeInfo db.NodeInfo
+			err = d.cluster.Transaction(func(tx *db.ClusterTx) error {
+				var err error
+				nodeInfo, err = tx.NodeByAddress(leader)
+				return err
+			})
+			if err != nil {
+				logger.Errorf("Failed to retrieve the information of leader node: %v", err)
+				return
+			}
+			fingerprints, err := d.cluster.ImagesGetByNodeID(nodeInfo.ID)
+			if err != nil {
+				logger.Errorf("Failed to retrieve the image fingerprints of leader node: %v", err)
+				return
+			}
+
+			imageImport := func(client lxd.ContainerServer, fingerprint string) error {
+				err := imageImportFromNode(filepath.Join(d.os.VarDir, "images"), client, fingerprint)
+				if err != nil {
+					return err
+				}
+
+				project := "default"
```

Comment:
Hmm, what is that?

It looks like you're getting a list of all images from the leader and then load them all into the new node but rather than associating them with the project they're used in, you're associating them all with the `default` project, this seems like it would lead to potential corruption.

Think of this case:
 - Two projects
 - No images in the `default` project
 - 10 images in the `blah` project
 - Join a second node with image sync enabled

All 10 images will now be synced and will be linked to the `default` project on the new server, which is incorrect and will lead to problems.

That's assuming that ImageAssociateNode indeed does something with the project information.

@freeekanayaka @adglkh